raiden/network/proxies/payment_channel.py

```diff
@@ -11,22 +12,40 @@ class PaymentChannel:
     def __init__(
         self,
         token_network: TokenNetwork,
-        partner_address: typing.Address,
+        channel_id: typing.ChannelID,
     ):
         self.token_network = token_network
-        self.partner_address = partner_address
+        self.channel_id = channel_id
+
+        # query the blockchain to get the partner addresses for the channel id
+        filter = token_network.proxy.contract.events.ChannelOpened.createFilter(
+            fromBlock=0,
```

Comment:
uhh, this can be bad, I think we should open an issue to fix this block

Reply:
What do you mean specifically? Insert the deloyment block of the contract here or the block where the channel was opened?

Reply:
> What do you mean specifically? 

Using `fromBlock=0` can be a performance bottle neck, therefor we should improve this.

>  Insert the deloyment block of the contract here or the block where the channel was opened?

Ideally we should use the block height at which the channel was created, but I don't think this is possible, since we are filtering for the events exactly because we want to know about the channel creating. I guess the token network block creation would be the easiest option.

note: I don't think we need to go overboard with this, the goal should be to add the required data as part of the proxy constructor